src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(({ sex, died }) => (
+    sex === 'm' && (
+      century
+        ? century === Math.ceil(died / 100)
+        : true
+    )));
+
+  const sumOfMenAges = men
+    .reduce((sum, { born, died }) => sum + (died - born), 0);
+
+  return divideAndRound(sumOfMenAges, men.length);
```

Comment:
But now `reduce` are duplicated, would be nice to move it to helper function (for example use both functions, and use function with reduce in first two functions and `divideAndRound`)